src/devices/nm-device.c

```diff
@@ -14283,7 +14284,9 @@ _cleanup_generic_post (NMDevice *self, CleanupType cleanup_type)
 		/* Check if the device was deactivated, and if so, delete_link.
 		 * Don't call delete_link synchronously because we are currently
 		 * handling a state change -- which is not reentrant. */
-		delete_on_deactivate_check_and_schedule (self, nm_device_get_ip_ifindex (self));
+		ifindex = nm_device_get_ip_ifindex (self);
+		if (ifindex > 0)
+			delete_on_deactivate_check_and_schedule (self, ifindex);
```

Comment:
wouldn't it be nicer if `delete_on_deactivate_check_and_schedule()` had this check for `ifindex > 0`? It already starts with a preamble of conditions when to do nothing.

Otherwise, there should be a `g_return_if_fail (ifindex > 0);` in `delete_on_deactivate_check_and_schedule()`.